src/tree/mod.rs

```diff
@@ -649,12 +647,16 @@ impl RatchetTree {
             self.blank_member(index);
             self.nodes[index.as_usize()] = leaf_node;
             if index == self.get_own_node_index() && !pending_kpbs.is_empty() {
-                let own_kpb = pending_kpbs
+                let own_kpb_index = match pending_kpbs
                     .iter()
-                    .find(|&kpb| kpb.get_key_package() == &update_proposal.key_package)
-                    .unwrap();
+                    .position(|kpb| kpb.get_key_package() == &update_proposal.key_package)
+                {
+                    Some(i) => i,
+                    None => panic!("Handle this error case"),
+                };
+                let own_kpb = pending_kpbs.remove(own_kpb_index);
```

Comment:
Here it's also not necessary to remove the keys. The parameter `pending_kpbs` is only non-empty when this function is called from `apply_commit.rs`.

Reply:
It's again necessary because we move it.